impl/src/main/java/io/kroxylicious/testing/kafka/invm/LoggingPrintStream.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.testing.kafka.invm;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.nio.charset.Charset;
+
+final class LoggingPrintStream {
+
+    static PrintStream loggingPrintStream(System.Logger logger, System.Logger.Level level) {
+        return new PrintStream(new LoggingOutputStream(logger, level));
+    }
+
+    private LoggingPrintStream() {
+    }
+
+    private static class LoggingOutputStream extends OutputStream {
+        private final ByteArrayOutputStream os = new ByteArrayOutputStream(1000);
+        private final System.Logger logger;
+        private final System.Logger.Level level;
+
+        public LoggingOutputStream(System.Logger logger, System.Logger.Level level) {
+            this.logger = logger;
+            this.level = level;
+        }
+
+        @Override
+        public void write(int b) {
```

Comment:
IIRC for performance reasons its recommended to [override](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/OutputStream.html#:~:text=Subclasses%20are%20encouraged%20to%20override%20this%20method%20and%20provide%20a%20more%20efficient%20implementation.) `write(bytes, start, length)` do we care enough in context?

If not it might be worth a comment.

Reply:
Given this class has one tiny use-case (the storage formatter), I'm not sure it is worth the effort.  If i were going to apply extra effort any place it would be Kafka. I'd raise a KIP suggesting a less opinionated interface for kafka.tools.StorageTool#formatCommand.  PrintStreams are weird beasts.

I'll add a comment.